app/src/test/js/syncCtrlTest.js

```diff
@@ -0,0 +1,81 @@
+describe('SyncCtrl', function() {
+
+    // Instantiate a new version of my module before each test
+    beforeEach(module('esbMessageAdminApp'));
+
+    var syncCtrl, http, globals, scope, rootScope, esbMessageService, filter;
+
+    beforeEach(inject(function($rootScope, $controller, $httpBackend, Globals,
+            EsbMessageService, $filter) {
+        rootScope = $rootScope;
+        scope = $rootScope.$new();
+        http = $httpBackend;
+        globals = Globals;
+        esbMessageService = EsbMessageService;
+        filter = $filter;
+
+        (syncCtrl) = $controller('SyncCtrl', {
+            '$scope' : scope
+        });
+
+        http.expectGET("api/key/tree/Entities")
+                .respond(entitiesSuccessResponse);
+        // Simulate a server response
+        http.flush();
+    }));
+
+    it('loads Sync Tab', function() {
+        expect(scope.entities).toEqual(entitiesSuccessResponse.tree);
+    });
+
+    it('selects key', function() {
+
+        scope.syncEntity = entitiesSuccessResponse.tree.children[0];
+        expect(scope.systems.length).toEqual(0);
+        scope.entityChange();
+        expect(scope.systems).toEqual(
+                entitiesSuccessResponse.tree.children[0].children);
+        scope.syncSystem = scope.systems[0];
+        expect(scope.keys.length).toEqual(0);
+        scope.systemChange();
```

Comment:
Similar to before, it is best practice to have more smaller tests than less tests that test multiple things. I would probably have separate tests for `entityChange()` and `systemChange()`